src/app/components/template-button/template-button.component.ts

```diff
@@ -8,6 +8,8 @@ import { Component, Input } from '@angular/core';
 export class TemplateButtonComponent {
     @Input() buttonText: string | undefined;
     @Input() buttonIcon: string | undefined;
+    @Output() buttonAction: EventEmitter<any> = new EventEmitter();
```

Comment:
Den expliziten Typen kannst du weglassen, wenn du das direkt mit einem Wer belegst (Geschmackssache).
Aber auf jeden Fall wäre es schon, den richtigen generischen Typen zu setzen, das wäre in dem Fall `void`.

```suggestion
    @Output() buttonAction = new EventEmitter<void>();
```